incubator/command-dump/src/test/java/io/aklivity/zilla/runtime/command/dump/internal/airline/ZillaDumpCommandTest.java

```diff
@@ -679,6 +683,144 @@ public void generateStreamsBuffer() throws Exception
+            .authority("localhost:7153")
+            .service("example.EchoService")
+            .method("EchoUnary")
+            .metadata("grpc-accept-encoding", "gzip")
+            .metadata("metadata-2", "hello")
+            .metadata("BASE64", "metadata-3", "4242")
+            .build());
+        BeginFW begin11 = beginRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+            .originId(0x000000090000001aL) // north_grpc_server
+            .routedId(0x000000090000001bL) // north_grpc_kafka_mapping
+            .streamId(0x0000000000000013L) // INI
+            .sequence(0)
+            .acknowledge(0)
+            .maximum(0)
+            .timestamp(0x0000000000000023L)
+            .traceId(0x0000000000000013L)
+            .affinity(0x0000000000000000L)
+            .extension(grpcBeginEx1, 0, grpcBeginEx1.capacity())
+            .build();
+        streams[0].write(BeginFW.TYPE_ID, begin11.buffer(), 0, begin11.sizeof());
+
+        DirectBuffer grpcBeginEx2 = new UnsafeBuffer(GrpcFunctions.beginEx()
+            .typeId(GRPC_TYPE_ID)
+            .scheme("http")
+            .authority("localhost:7153")
+            .service("example.EchoService")
+            .method("EchoUnary")
+            .metadata("long field", "Z".repeat(200))
+            .metadata("metadata-2", "hello")
+            .build());
+        BeginFW begin12 = beginRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+            .originId(0x000000090000001aL) // north_grpc_server
+            .routedId(0x000000090000001bL) // north_grpc_kafka_mapping
+            .streamId(0x0000000000000012L) // REP
+            .sequence(0)
+            .acknowledge(0)
+            .maximum(0)
+            .timestamp(0x0000000000000024L)
+            .traceId(0x0000000000000013L)
+            .affinity(0x0000000000000000L)
+            .extension(grpcBeginEx2, 0, grpcBeginEx2.capacity())
+            .build();
+        streams[0].write(BeginFW.TYPE_ID, begin12.buffer(), 0, begin12.sizeof());
+
+        // data frame with extension, without payload, payload length is -1
+        DirectBuffer grpcDataEx1 = new UnsafeBuffer(new byte[]{GRPC_TYPE_ID, 0, 0, 0, 42, 0, 0, 0});
+        DataFW data6 = dataRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+            .originId(0x000000090000001aL) // north_grpc_server
+            .routedId(0x000000090000001bL) // north_grpc_kafka_mapping
+            .streamId(0x0000000000000013L) // INI
+            .sequence(0)
+            .acknowledge(0)
+            .maximum(0)
+            .timestamp(0x0000000000000025L)
+            .traceId(0x0000000000000013L)
+            .budgetId(0x0000000000000013L)
+            .reserved(0x00000042)
+            .extension(grpcDataEx1, 0, grpcDataEx1.capacity())
+            .build();
+        streams[0].write(DataFW.TYPE_ID, data6.buffer(), 0, data6.sizeof());
+
+        // data frame with extension, without payload, payload length is 0
+        DirectBuffer grpcDataPayload1 = new UnsafeBuffer(EMPTY_BYTES);
```

Comment:
Let's use no-args constructor `new UnsafeBuffer()` and remove `EMPTY_BYTES` constant.

Reply:
fixed